zrml/neo-swaps/src/utility.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright 2024 Forecasting Technologies LTD.
+//
+// This file is part of Zeitgeist.
+//
+// Zeitgeist is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at
+// your option) any later version.
+//
+// Zeitgeist is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+
+use sp_runtime::SaturatedConversion;
+
+pub(crate) trait LogCeil {
+    fn log_ceil(&self) -> Self;
+}
+
+impl LogCeil for u16 {
+    fn log_ceil(&self) -> Self {
+        let x = *self;
```

Comment:
Should we handle the `x = 0` case? If I understand it correctly, it would currently return `1`. I don't mind if we leave it like this, since more weight (is used for combinatorial pool extrinsic weights) is always "safer".